src/ts/components/subtitleselectbox.ts

```diff
@@ -51,4 +54,18 @@ export class SubtitleSelectBox extends SelectBox {
     // Populate subtitles at startup
     updateSubtitles();
   }
+
+  /**
+   * Add a css class when subtitles are activated
+   * @param {boolean} activated
+   */
+  private setSubtitleActivatedClass(activated: boolean): void {
```

Comment:
Why is this necessary for the subtitles but not for the audio tracks? Wouldn't this be better placed in the `SelectBox` base class? Is this even necessary or can't the `SelectBox` be natively styled as needed?

Also it seems that you need this only for styling in the demo style file. Please don't build functionality into the core components that is only used for a specific demo, i.e. you're implementing functionality here but styling within the demo file. That does not fit together. When implementing something into UI components, it must always be generic and be of use to all users of the framework, or added externally and specifically for the use case.

Reply:
We wanted to display the 'on-state' icon when a subtitle is selected.
As far as i understand there is always an audio track but you can turn off / on subtitles.

For now it is only used in the demo style file but we could use it too in the future. I think it would be nice to have an indicator in the ui when subtitles are active.

I will have a look at this use-case again in #121

Reply:
Then why is this implemented in the `SubtitleSelectBox`? Shouldn't this be a feature of the `SettingsToggleButton`?